### DISCUSSION TITLE
Login/Signup

### DISCUSSION THREAD

**Discussion by YunusEmreAlps** [Planet-NULLCAST/curriculum#878] [Q&A]
Hi, 
I hope you're well.

I keep getting "Request failed with status code 404" error during login and registration processes. What do I need to do to complete the login process? Can you explain the login module for me?

Thanks for this project. I learned so many things from you. I'm gonna change, curriculum editor with this https://github.com/enesozturk/awesome-course project editor.

All the best,
Have a good day

**Comment by that-coder** [comment]
Hello Yunus

Thanks for trying out our curriculum, and good to hear your feedback. Our team will get back to you once the issue is fixed.

We are welcoming your contributions to the curriculum/ community as well. Meanwhile, you can join our discord channel to connect with other awesome devs around. 

Join link: https://discord.gg/6s2qTkQe

Happy Hacking!

**Comment by thealpha93** [comment]
@YunusEmreAlps 

Give us a day. Let me see what's wrong with our registration system.
Thanks

**Comment by thealpha93** [comment]
@YunusEmreAlps 
It seems you are registered with nullcast. And you haven't verified the email address. 

Can you check for a verification mail from nullcast?
Also can you give me more details as to which API returned the 404?
Thanks

**Comment by thealpha93** [comment]
@YunusEmreAlps 

I have verified your account internally. There might be an issue with our email server.

can you check if you can log in again?
THanks

**Comment by YunusEmreAlps** [comment]
First of all, I tried to enter the "https://nullcast.dev/" address to see the design and operation of the system. However, the mail did not come.
- ![Screenshot (367)](https://user-images.githubusercontent.com/44679555/161428090-a3e5e8a5-fd76-450b-b4d7-304b4c09932e.png)
- ![Screenshot (368)](https://user-images.githubusercontent.com/44679555/161428172-56e77ca2-2524-4b7d-857d-6bd323d4006c.png)

Then I downloaded the code and ran it in my environment, but I still couldn't log in.
Do I need to download another project for the backend?

- ![Screenshot (369)](https://user-images.githubusercontent.com/44679555/161428492-7f75e3db-7074-414d-aef5-c4be90f50ba0.png)
- ![Screenshot (370)](https://user-images.githubusercontent.com/44679555/161428589-8ccda8e2-d450-4c0a-8d35-83370a4b64f6.png)
- ![Screenshot (371)](https://user-images.githubusercontent.com/44679555/161428626-8bcad220-d54d-4230-ae21-af729372bee1.png)

"XHR failed loading: POST "http://localhost:3000/undefined/api/v1/user"." 

I want to integrate different editor and Multiple auth (Google, GitHub) for this system. With this project, I want to learn and practice React.

Thank you @thealpha93  and @that-coder. By the way, the community is awesome.
Have a good day 🖐🖐🖐

**Reply by thealpha93** [reply]
@YunusEmreAlps 

Yes you need to run a separate backend for this project.
Unfortunately that repo is private.
But I can give you access to our dev server.

Change the contents of `.env` to this

```
#development
BASE_URL=https://test-api.nullcast.dev
CLIENT_URL=https://test-app.nullcast.dev
EDITOR_URL=https://test-editor.nullcast.dev
BUCKET_URL=https://n8o9asugsb.execute-api.us-east-1.amazonaws.com
SERVER_URL=http://test-api.nullcast.dev
```

**Reply by thealpha93** [reply]
You're welcome to integrate multiple auth and a different editor for this project.
Reach out to me on discord for any clarifications.
THanks
